extensions/vertx-http/deployment-spi/pom.xml

```diff
@@ -18,6 +18,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-core-deployment</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>io.vertx</groupId>
+            <artifactId>vertx-web</artifactId>
+        </dependency>
```

Comment:
Do we really want to add this dependency? The idea would be to not depend on anything Vert.x right?

Reply:
I don't know if it's necessary, but as far as I'm concerned, it's fine. A mandatory dependency in the deployment artifact is acceptable as long as it can be optional at runtime (and from what I see in Marko 's PR, it can be).

Reply:
(to clarify, the initial goal was to make the dependency optional at runtime, though Clément may have gone a bit further here)